aidants_connect_carto/apps/core/search.py

```diff
@@ -91,15 +134,23 @@ def _build_results(self):
         to the calling code, typically as a more or less complex `dict`, which
         could also contain meta-information about the search, like the total
         number of results, ...
+        - order_by
+        - opening_hours
+        - pagination
         """
-        page_number = self.query.get("page", 1)
-        offset = self.RESULTS_PER_PAGE * (page_number - 1)
+        # order_by
+        order_by_field = self.query.get("order_by", self.DEFAULT_ORDER_BY)
+        self.queryset = self.queryset.order_by(order_by_field)
+
+        raw_results = self.queryset
 
-        raw_results = self.queryset[
-            offset : offset + self.RESULTS_PER_PAGE
-        ]  # <-- db query is executed here!
+        # opening_hours (place.is_open) filter
+        if self.query.get("opening_hours", "") != "":
+            raw_results = [place for place in raw_results if place.is_open]
```

Comment:
Alors, _oui mais_ :)

Ça va marcher, très bien pour le moment, et du coup j'ai envie de dire qu'à ce stade (prototypage, itérations rapides pour jauger de la pertinence du concept, ...) c'est parfaitement ok, mais de manière sous-jacente ça pose des problèmes, et du coup je profite de l'occasion pour une petite explication anticipée ^^

On est dans une situation classique de recherche, dans laquelle on peut exécuter des actions à deux endroits distincts : 1. dans la base de données, en ciselant une requête SQL (ce qui est le rôle de l'ORM : nous donner un langage expressif pour nous permettre de ne pas faire de SQL _directement_), et/ou 2. dans le code Python lui-même, une fois le résultat renvoyé par la base de données.

L'idée générale, pour avoir de bonnes performances, est de _faire faire un maximum de travail par la base de données_ — après tout, elle est là pour ça, elle est codée dans un langage plus performant, et ses algos de recherche sont optimisés depuis des années par une palanquée de gens très très compétents 😁 

Or, là, `if place.is_open` est fait "en Python", ce qui implique que la base elle-même _n'aura pas fait ce filtrage en amont_, et donc potentiellement aura fait transiter "pour rien" sur le réseau toutes les informations sur les lieux pour lesquels c'est faux, ce qui a un impact en termes de performances directement proportionnel à la volumétrie du jeu de données (donc au début avec 3 lignes de test, on ne s'en rend pas compte, mais quand on commence à en avoir des milliers... ça raaaaaame :p).

Là encore, c'est clairement prématuré de se préoccuper de ça maintenant, mais je pense que c'est une bonne occasion pour te/vous sensibiliser au sujet — ping @Anna-Livia aussi au passage :)

Reply:
Je suis tout à fait d'accord, je l'ai rajouté pour le POC, il va falloir qu'on trouve une autre solution à terme. Car quid de récupérer X lieux (la taille de la page), pour ensuite se rendre compte qu'aucun n'est ouvert, refaire la requête ?

Reply:
Exactement, c'est le genre de "détails" où peut se cacher beaucoup de complexité d'optimisation. Le mieux serait probablement de se faire un petit atelier à ce sujet quand on pourra ^^